packages/server/src/schema/resolvers.ts

```diff
@@ -20,12 +21,19 @@ export interface CreateSimulationParameters {
 }
 
 export const createSimulation: Resolver<CreateSimulationParameters, SimulationState> = {
-  resolve({ simulator, options = {} }, ctx) {
+  async resolve({ simulator, options = {} }, ctx) {
     let { atom, scope, newid } = ctx;
 
-    let id = newid();
+    let id = options.key ?? newid();
+
     let simulation = atom.slice("simulations", id);
 
+    if(!!simulation.get()) {
+      simulation.remove();
+    }
```

Comment:
Do we still need this functionality now that simulations are being fully destroyed after each test? I can still see having this as a feature, but maybe as a separate PR. 

In both cases though, we need to do the whole dance here that we do in `destroySimulation()` to make sure that the old simulation is torn down fully before we create the new one.

If we do move forward with this feature, I was thinking we should call it `uniqueKey` or `singletonKey`